recipes/restinio/all/conanfile.py

```diff
@@ -64,51 +59,44 @@ def package_id(self):
         self.info.clear()
 
     def validate(self):
-        minimal_cpp_standard = "14"
-        if self.settings.get_safe("compiler.cppstd"):
+        minimal_cpp_standard = "17"
+        if self.settings.compiler.get_safe("cppstd"):
             check_min_cppstd(self, minimal_cpp_standard)
+
         minimal_version = {
-            "gcc": "5",
-            "clang": "3.4",
-            "apple-clang": "10",
+            "gcc": "9",
+            "clang": "10",
+            "apple-clang": "11",
+            "Visual Studio": "17",
+            "msvc": "191"
         }
-        check_min_vs(self, 190)
-        if not is_msvc(self):
-            minimum_version = minimal_version.get(str(self.info.settings.compiler), False)
-            if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
-                raise ConanInvalidConfiguration(
-                    f"{self.ref} requires C++{minimal_cpp_standard}, which your compiler does not support."
-                )
-
-    def generate(self):
-        tc = CMakeToolchain(self)
-        tc.variables["RESTINIO_INSTALL"] = True
-        tc.variables["RESTINIO_FIND_DEPS"] = False
-        tc.variables["RESTINIO_USE_EXTERNAL_EXPECTED_LITE"] = True
-        tc.variables["RESTINIO_USE_EXTERNAL_OPTIONAL_LITE"] = True
-        tc.variables["RESTINIO_USE_EXTERNAL_STRING_VIEW_LITE"] = True
-        tc.variables["RESTINIO_USE_EXTERNAL_VARIANT_LITE"] = True
-        tc.generate()
-
-        deps = CMakeDeps(self)
-        deps.generate()
+
+        compiler = str(self.settings.compiler)
+        if compiler not in minimal_version:
+            self.output.warning(
+                "%s recipe lacks information about the %s compiler standard version support" % (self.name, compiler))
+            self.output.warning(
+                "%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
+            return
+
+        version = Version(self.settings.compiler.version)
+        if version < minimal_version[compiler]:
+            raise ConanInvalidConfiguration("%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
+
+    def build(self):
+        pass
 
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def package(self):
+        copy(self, "*.*pp", src=os.path.join(self.source_folder, "dev", "restinio"), dst=os.path.join(self.package_folder, "include", "restinio"))
         copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
```

Comment:
Is CMAKE.install() not suitable anymore?

Reply:
I could have use it, yes, but it goes with maintenance costs:
* set the tuning variables right (they can be added/removed and their names/values might be changed in future versions) 
* Conan itself may change the cmake-routines (like v1=>v2).

On the other hand RESTinio takes a promise that a set of headers would be enough - so copying sources is a nice way to go without extra cmake machinery.